[BUG]  sinkService crashes (stack smash) when updating config.

**Describe the bug**
We noticed that when we attempt to configure sinks, the wirepas sink services are crashing. The crash also causes a stack smash, so this is also a security vulnerability.

**To Reproduce**
Steps to reproduce the behavior:
1. Start service ...
2. configure_node.py -s sink2 -n 5574241 -r "sink csma-ca"  -N 6566628 -c 8 -ak FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF -ck FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF -S true set
3. Could not evaluate protobuf implementation type
Waiting for sink sink2 to be ready
Checking dbus system service com.wirepas.sink.sink2
Service com.wirepas.sink.sink2 is up.
Waiting for sink sink2 to be ready
Checking dbus system service com.wirepas.sink.sink2
Service com.wirepas.sink.sink2 is up.
2024-09-11 10:47:05,722 | [INFO] msap:Start request result = 0x00
2024-09-11 10:47:05,805 | [INFO] Config:Stack started manually
*** stack smashing detected ***: terminated
ERROR:root:Could not get network address
Configuration done with result = GatewayResultCode.GW_RES_OK



**Expected behavior**
The sinkService should not crash. It should update the configuration.

**Platform details:**
 - OS: Poky 6.6.23-linux4microchip-2024.04-rc1-g468020704fb5-dirty
 - Gateway host: Microchip Sama5d2-ptc-ek-sd:
 - Python version: 3.10.14
 - Docker engine:n/a
 - MQTT Broker vendor:mosquitto

**Transport service settings:**
n/a
**Sink service settings:**
root@sama5d2-ptc-ek-sd:~# cat /lib/systemd/system/wirepas-sink-ttys2.service
[Unit]
Description=Wirepas Sink 2 Service
StartLimitIntervalSec=500
StartLimitBurst=5

[Service]
Type=dbus
BusName=com.wirepas.sink.sink2
User=wirepas
ExecStartPre=/bin/sh -c "stty -F /dev/ttyS5 raw; stty -F /dev/ttyS5 -echo"
ExecStart=/usr/bin/sinkService -p /dev/ttyS5 -i 2
StandardOutput=journal
Restart=on-failure
RestartSec=5s

[Install]


**Screenshots**

**Additional context**
To confirm this crash, I stopped the sink services and ran them manually from a terminal. That allowed me to collect the sink service debug prints, the stack smash print and the exit for the forked service pid.

root@sama5d2-ptc-ek-sd:~# su -s "/bin/bash" -c '/usr/bin/sinkService -p /dev/ttyS5 -i 2' wirepas &
[1] 18572
root@sama5d2-ptc-ek-sd:~# 2024-09-11 10:43:35,915 | [INFO] Main:Starting Sink service:
	-Port is /dev/ttyS5
	-Baudrate is 0
	-Dbus Service name is com.wirepas.sink.sink2
2024-09-11 10:43:35,929 | [INFO] Main:Auto baudrate: testing 125000 bps
2024-09-11 10:43:35,932 | [DEBUG] SERIAL:Custom bitrate set: 125000
2024-09-11 10:43:35,938 | [INFO] wpc_int:WPC initialized
2024-09-11 10:43:36,482 | [ERROR] wpc_int:Didn't receive answer to the request 0x0e error is: -2
2024-09-11 10:43:37,032 | [ERROR] wpc_int:Didn't receive answer to the request 0x0e error is: -2
2024-09-11 10:43:37,033 | [WARNING] linux_plat:Exiting dispatch thread
2024-09-11 10:43:37,582 | [ERROR] wpc_int:Didn't receive answer to the request 0x04 error is: -2
2024-09-11 10:43:37,582 | [ERROR] wpc_int:No poll answer
2024-09-11 10:43:37,583 | [WARNING] linux_plat:Exiting polling thread
2024-09-11 10:43:37,583 | [INFO] Main:Auto baudrate: testing 115200 bps
2024-09-11 10:43:37,585 | [DEBUG] SERIAL:Custom bitrate set: 115200
2024-09-11 10:43:37,587 | [INFO] wpc_int:WPC initialized
2024-09-11 10:43:37,593 | [INFO] Main:Node is running mesh API version 16 (uart baudrate is 115200 bps)
2024-09-11 10:43:37,593 | [INFO] Main:Uart baudrate found: 115200 bps
2024-09-11 10:43:37,629 | [INFO] Config:Stack version is: 5.2.0.53
2024-09-11 10:43:37,633 | [INFO] Config:Stack is stopped
2024-09-11 10:43:38,091 | [INFO] msap:Status is 0x01
2024-09-11 10:43:38,139 | [INFO] Config:Stack version is: 5.2.0.53

root@sama5d2-ptc-ek-sd:~# 
root@sama5d2-ptc-ek-sd:~# 
root@sama5d2-ptc-ek-sd:~# dbus-send --system --dest=org.freedesktop.DBus --type=method_call --print-reply / org.freedesktop.DBus.ListNames | grep wire
      string "com.wirepas.sink.sink1"
      string "com.wirepas.sink.sink2"

root@sama5d2-ptc-ek-sd:~# su -s "/bin/bash" -c 'python3 /usr/bin/configure_node.py -s sink2 -n 5574241 -r "sink csma-ca"  -N 6566628 -c 8 -ak FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF -ck FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF -S true set' wirepas
Could not evaluate protobuf implementation type

2024-09-11 10:47:05,722 | [INFO] msap:Start request result = 0x00
2024-09-11 10:47:05,805 | [INFO] Config:Stack started manually
*** stack smashing detected ***: terminated

ERROR:root:Could not get network address
Configuration done with result = GatewayResultCode.GW_RES_OK

[1]-  Done(134)               su -s "/bin/bash" -c '/usr/bin/sinkService -p /dev/ttyS5 -i 2' wirepas


Here is the same sequence with dbus-monitor showing the stack smash crash happens when the stop stack is initiated.

root@sama5d2-ptc-ek-sd:~# su -s "/bin/bash" -c '/usr/bin/sinkService -p /dev/ttyS5 -i 2' wirepas &
[3] 2242
root@sama5d2-ptc-ek-sd:~# 2024-09-11 11:01:35,708 | [INFO] Main:Starting Sink service:
	-Port is /dev/ttyS5
	-Baudrate is 0
	-Dbus Service name is com.wirepas.sink.sink2
2024-09-11 11:01:35,712 | [INFO] Main:Auto baudrate: testing 125000 bps
2024-09-11 11:01:35,714 | [DEBUG] SERIAL:Custom bitrate set: 125000
2024-09-11 11:01:35,717 | [INFO] wpc_int:WPC initialized
2024-09-11 11:01:36,272 | [ERROR] wpc_int:Didn't receive answer to the request 0x0e error is: -2
2024-09-11 11:01:36,823 | [ERROR] wpc_int:Didn't receive answer to the request 0x0e error is: -2
2024-09-11 11:01:36,824 | [WARNING] linux_plat:Exiting dispatch thread
2024-09-11 11:01:37,373 | [ERROR] wpc_int:Didn't receive answer to the request 0x04 error is: -2
2024-09-11 11:01:37,380 | [ERROR] wpc_int:No poll answer
2024-09-11 11:01:37,382 | [WARNING] linux_plat:Exiting polling thread
2024-09-11 11:01:37,383 | [INFO] Main:Auto baudrate: testing 115200 bps
2024-09-11 11:01:37,383 | [DEBUG] SERIAL:Custom bitrate set: 115200
2024-09-11 11:01:37,385 | [INFO] wpc_int:WPC initialized
2024-09-11 11:01:37,392 | [INFO] Main:Node is running mesh API version 16 (uart baudrate is 115200 bps)
2024-09-11 11:01:37,392 | [INFO] Main:Uart baudrate found: 115200 bps
2024-09-11 11:01:37,429 | [INFO] Config:Stack version is: 5.2.0.53
2024-09-11 11:01:37,433 | [INFO] Config:Stack is started

root@sama5d2-ptc-ek-sd:~# 
root@sama5d2-ptc-ek-sd:~# 
root@sama5d2-ptc-ek-sd:~# dbus-monitor --system &
[4] 2433
root@sama5d2-ptc-ek-sd:~# signal time=1726052508.392829 sender=org.freedesktop.DBus -> destination=:1.9664 serial=2 path=/org/freedesktop/DBus; interface=org.freedesktop.DBus; member=NameAcquired
   string ":1.9664"
signal time=1726052508.398109 sender=org.freedesktop.DBus -> destination=:1.9664 serial=4 path=/org/freedesktop/DBus; interface=org.freedesktop.DBus; member=NameLost
   string ":1.9664"
signal time=1726052508.405709 sender=org.freedesktop.DBus -> destination=(null destination) serial=19352 path=/org/freedesktop/DBus; interface=org.freedesktop.DBus; member=NameOwnerChanged
   string ":1.9664"
   string ":1.9664"
   string ""
root@sama5d2-ptc-ek-sd:~# su -s "/bin/bash" -c 'python3 /usr/bin/configure_node.py -s sink2 -n 5574241 -r "sink csma-ca"  -N 6566628 -c 8 -ak FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF -ck FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF -S true set' wirepas





method call time=1726052526.887396 sender=:1.9672 -> destination=com.wirepas.sink.sink2 serial=17 path=/com/wirepas/sink; interface=org.freedesktop.DBus.Properties; member=Get
   string "com.wirepas.sink.config1"
   string "StackStatus"
method return time=1726052526.897800 sender=:1.9663 -> destination=:1.9672 serial=39 reply_serial=17
   variant       byte 0

method call time=1726052526.919685 sender=:1.9672 -> destination=com.wirepas.sink.sink2 serial=18 path=/com/wirepas/sink; interface=org.freedesktop.DBus.Properties; member=Get
   string "com.wirepas.sink.config1"
   string "StackStatus"
method return time=1726052526.933837 sender=:1.9663 -> destination=:1.9672 serial=40 reply_serial=18
   variant       byte 0

method call time=1726052526.943021 sender=:1.9672 -> destination=com.wirepas.sink.sink2 serial=19 path=/com/wirepas/sink; interface=com.wirepas.sink.config1; member=SetStackState
   boolean false

2024-09-11 11:02:06,966 | [INFO] msap:Stop request result = 0x00
2024-09-11 11:02:07,470 | [INFO] Config:Stack stopped manually
*** stack smashing detected ***: terminated

signal time=1726052527.503938 sender=:1.9663 -> destination=(null destination) serial=41 path=/com/wirepas/sink; interface=com.wirepas.sink.config1; member=StackStopped

method return time=1726052527.547014 sender=:1.9663 -> destination=:1.9672 serial=42 reply_serial=19
   boolean true

signal time=1726052527.549516 sender=org.freedesktop.DBus -> destination=:1.9663 serial=6 path=/org/freedesktop/DBus; interface=org.freedesktop.DBus; member=NameLost
   string "com.wirepas.sink.sink2"

signal time=1726052527.559861 sender=org.freedesktop.DBus -> destination=(null destination) serial=19369 path=/org/freedesktop/DBus; interface=org.freedesktop.DBus; member=NameOwnerChanged
   string "com.wirepas.sink.sink2"
   string ":1.9663"
   string ""

signal time=1726052527.564136 sender=org.freedesktop.DBus -> destination=:1.9663 serial=7 path=/org/freedesktop/DBus; interface=org.freedesktop.DBus; member=NameLost
   string ":1.9663"

signal time=1726052527.573742 sender=org.freedesktop.DBus -> destination=(null destination) serial=19370 path=/org/freedesktop/DBus; interface=org.freedesktop.DBus; member=NameOwnerChanged
   string ":1.9663"
   string ":1.9663"
   string ""

method call time=1726052527.588572 sender=:1.28 -> destination=com.wirepas.sink.sink2 serial=2197 path=/com/wirepas/sink; interface=org.freedesktop.DBus.Properties; member=Get
   string "com.wirepas.sink.config1"
   string "StackStatus"
error time=1726052527.592931 sender=org.freedesktop.DBus -> destination=:1.28 error_name=org.freedesktop.DBus.Error.ServiceUnknown reply_serial=2197
   string "The name com.wirepas.sink.sink2 was not provided by any .service files"

method call time=1726052527.620012 sender=:1.9672 -> destination=com.wirepas.sink.sink2 serial=20 path=/com/wirepas/sink; interface=org.freedesktop.DBus.Properties; member=Set
   string "com.wirepas.sink.config1"
   string "NodeAddress"
   variant       uint32 5574241
error time=1726052527.624281 sender=org.freedesktop.DBus -> destination=:1.9672 error_name=org.freedesktop.DBus.Error.ServiceUnknown reply_serial=20
   string "The name com.wirepas.sink.sink2 was not provided by any .service files"

method call time=1726052527.641923 sender=:1.28 -> destination=com.wirepas.sink.sink2 serial=2198 path=/com/wirepas/sink; interface=org.freedesktop.DBus.Properties; member=Get
   string "com.wirepas.sink.config1"
   string "NodeAddress"
error time=1726052527.662850 sender=org.freedesktop.DBus -> destination=:1.28 error_name=org.freedesktop.DBus.Error.ServiceUnknown reply_serial=2198
   string "The name com.wirepas.sink.sink2 was not provided by any .service files"

method call time=1726052527.672643 sender=:1.28 -> destination=com.wirepas.sink.sink2 serial=2199 path=/com/wirepas/sink; interface=org.freedesktop.DBus.Properties; member=Get
   string "com.wirepas.sink.config1"
   string "NodeRole"
error time=1726052527.701100 sender=org.freedesktop.DBus -> destination=:1.28 error_name=org.freedesktop.DBus.Error.ServiceUnknown reply_serial=2199
   string "The name com.wirepas.sink.sink2 was not provided by any .service files"

method call time=1726052527.752609 sender=:1.28 -> destination=com.wirepas.sink.sink2 serial=2200 path=/com/wirepas/sink; interface=org.freedesktop.DBus.Properties; member=Get
   string "com.wirepas.sink.config1"
   string "NetworkAddress"
error time=1726052527.766772 sender=org.freedesktop.DBus -> destination=:1.28 error_name=org.freedesktop.DBus.Error.ServiceUnknown reply_serial=2200
   string "The name com.wirepas.sink.sink2 was not provided by any .service files"

ERROR:root:Cannot set 5574241 for param node_address on sink sink2: GW_RES_INTERNAL_ERROR (g-dbus-error-quark: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name com.wirepas.sink.sink2 was not provided by any .service files (2))

method call time=1726052527.787728 sender=:1.28 -> destination=com.wirepas.sink.sink2 serial=2201 path=/com/wirepas/sink; interface=org.freedesktop.DBus.Properties; member=Get
   string "com.wirepas.sink.config1"
   string "NetworkChannel"
error time=1726052527.799758 sender=org.freedesktop.DBus -> destination=:1.28 error_name=org.freedesktop.DBus.Error.ServiceUnknown reply_serial=2201
   string "The name com.wirepas.sink.sink2 was not provided by any .service files"

method call time=1726052527.815201 sender=:1.28 -> destination=com.wirepas.sink.sink2 serial=2202 path=/com/wirepas/sink; interface=org.freedesktop.DBus.Properties; member=Get
   string "com.wirepas.sink.config1"
   string "ChannelMap"
error time=1726052527.825121 sender=org.freedesktop.DBus -> destination=:1.28 error_name=org.freedesktop.DBus.Error.ServiceUnknown reply_serial=2202
   string "The name com.wirepas.sink.sink2 was not provided by any .service files"

ERROR:root:Cannot set 17 for param node_role on sink sink2: GW_RES_INTERNAL_ERROR (g-dbus-error-quark: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name com.wirepas.sink.sink2 was not provided by any .service files (2))

method call time=1726052527.849511 sender=:1.9672 -> destination=com.wirepas.sink.sink2 serial=21 path=/com/wirepas/sink; interface=org.freedesktop.DBus.Properties; member=Get
   string "com.wirepas.sink.config1"
   string "StackStatus"
error time=1726052527.853527 sender=org.freedesktop.DBus -> destination=:1.9672 error_name=org.freedesktop.DBus.Error.ServiceUnknown reply_serial=21
   string "The name com.wirepas.sink.sink2 was not provided by any .service files"

method call time=1726052527.857086 sender=:1.28 -> destination=com.wirepas.sink.sink2 serial=2203 path=/com/wirepas/sink; interface=org.freedesktop.DBus.Properties; member=Get
   string "com.wirepas.sink.config1"
   string "ACRangeMax"
error time=1726052527.862433 sender=org.freedesktop.DBus -> destination=:1.28 error_name=org.freedesktop.DBus.Error.ServiceUnknown reply_serial=2203
   string "The name com.wirepas.sink.sink2 was not provided by any .service files"

method call time=1726052527.894592 sender=:1.28 -> destination=com.wirepas.sink.sink2 serial=2204 path=/com/wirepas/sink; interface=org.freedesktop.DBus.Properties; member=Get
   string "com.wirepas.sink.config1"
   string "ACRangeMaxCur"
error time=1726052527.899131 sender=org.freedesktop.DBus -> destination=:1.28 error_name=org.freedesktop.DBus.Error.ServiceUnknown reply_serial=2204
   string "The name com.wirepas.sink.sink2 was not provided by any .service files"

ERROR:root:Cannot set 6566628 for param network_address on sink sink2: GW_RES_INTERNAL_ERROR (g-dbus-error-quark: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name com.wirepas.sink.sink2 was not provided by any .service files (2))

method call time=1726052527.944163 sender=:1.9672 -> destination=com.wirepas.sink.sink2 serial=22 path=/com/wirepas/sink; interface=org.freedesktop.DBus.Properties; member=Get
   string "com.wirepas.sink.config1"
   string "StackStatus"
error time=1726052527.953731 sender=org.freedesktop.DBus -> destination=:1.9672 error_name=org.freedesktop.DBus.Error.ServiceUnknown reply_serial=22
   string "The name com.wirepas.sink.sink2 was not provided by any .service files"

method call time=1726052527.964579 sender=:1.28 -> destination=com.wirepas.sink.sink2 serial=2205 path=/com/wirepas/sink; interface=org.freedesktop.DBus.Properties; member=Get
   string "com.wirepas.sink.config1"
   string "ChRangeMax"
error time=1726052527.974230 sender=org.freedesktop.DBus -> destination=:1.28 error_name=org.freedesktop.DBus.Error.ServiceUnknown reply_serial=2205
   string "The name com.wirepas.sink.sink2 was not provided by any .service files"

method call time=1726052527.986631 sender=:1.28 -> destination=com.wirepas.sink.sink2 serial=2206 path=/com/wirepas/sink; interface=org.freedesktop.DBus.Properties; member=Get
   string "com.wirepas.sink.config1"
   string "MaxMtu"
error time=1726052528.003416 sender=org.freedesktop.DBus -> destination=:1.28 error_name=org.freedesktop.DBus.Error.ServiceUnknown reply_serial=2206
   string "The name com.wirepas.sink.sink2 was not provided by any .service files"

method call time=1726052528.014717 sender=:1.9672 -> destination=com.wirepas.sink.sink2 serial=23 path=/com/wirepas/sink; interface=org.freedesktop.DBus.Properties; member=Get
   string "com.wirepas.sink.config1"
   string "StackStatus"
error time=1726052528.022470 sender=org.freedesktop.DBus -> destination=:1.9672 error_name=org.freedesktop.DBus.Error.ServiceUnknown reply_serial=23
   string "The name com.wirepas.sink.sink2 was not provided by any .service files"

ERROR:root:Cannot set 8 for param network_channel on sink sink2: GW_RES_INTERNAL_ERROR (g-dbus-error-quark: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name com.wirepas.sink.sink2 was not provided by any .service files (2))

method call time=1726052528.044177 sender=:1.28 -> destination=com.wirepas.sink.sink2 serial=2207 path=/com/wirepas/sink; interface=org.freedesktop.DBus.Properties; member=Get
   string "com.wirepas.sink.config1"
   string "HwMagic"
error time=1726052528.048537 sender=org.freedesktop.DBus -> destination=:1.28 error_name=org.freedesktop.DBus.Error.ServiceUnknown reply_serial=2207
   string "The name com.wirepas.sink.sink2 was not provided by any .service files"

method call time=1726052528.064102 sender=:1.9672 -> destination=com.wirepas.sink.sink2 serial=24 path=/com/wirepas/sink; interface=org.freedesktop.DBus.Properties; member=Get
error time=1726052528.079318 sender=org.freedesktop.DBus -> destination=:1.9672 error_name=org.freedesktop.DBus.Error.ServiceUnknown reply_serial=24
   string "The name com.wirepas.sink.sink2 was not provided by any .service files"
   string "com.wirepas.sink.config1"
   string "StackStatus"

ERROR:root:Cannot set bytearray(b'\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff') for param authentication_key on sink sink2: GW_RES_INTERNAL_ERROR (g-dbus-error-quark: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name com.wirepas.sink.sink2 was not provided by any .service files (2))

method call time=1726052528.093687 sender=:1.28 -> destination=com.wirepas.sink.sink2 serial=2208 path=/com/wirepas/sink; interface=org.freedesktop.DBus.Properties; member=Get
   string "com.wirepas.sink.config1"
   string "StackProfile"
error time=1726052528.101783 sender=org.freedesktop.DBus -> destination=:1.28 error_name=org.freedesktop.DBus.Error.ServiceUnknown reply_serial=2208
   string "The name com.wirepas.sink.sink2 was not provided by any .service files"

method call time=1726052528.132660 sender=:1.28 -> destination=com.wirepas.sink.sink2 serial=2209 path=/com/wirepas/sink; interface=org.freedesktop.DBus.Properties; member=Get
   string "com.wirepas.sink.config1"
   string "FirmwareVersion"
error time=1726052528.151021 sender=org.freedesktop.DBus -> destination=:1.28 error_name=org.freedesktop.DBus.Error.ServiceUnknown reply_serial=2209
   string "The name com.wirepas.sink.sink2 was not provided by any .service files"

method call time=1726052528.154754 sender=:1.9672 -> destination=com.wirepas.sink.sink2 serial=25 path=/com/wirepas/sink; interface=org.freedesktop.DBus.Properties; member=Get
   string "com.wirepas.sink.config1"
   string "StackStatus"
error time=1726052528.181633 sender=org.freedesktop.DBus -> destination=:1.9672 error_name=org.freedesktop.DBus.Error.ServiceUnknown reply_serial=25
   string "The name com.wirepas.sink.sink2 was not provided by any .service files"

method call time=1726052528.189085 sender=:1.28 -> destination=com.wirepas.sink.sink2 serial=2210 path=/com/wirepas/sink; interface=org.freedesktop.DBus.Properties; member=Get
   string "com.wirepas.sink.config1"
   string "AppConfigMaxSize"
error time=1726052528.205606 sender=org.freedesktop.DBus -> destination=:1.28 error_name=org.freedesktop.DBus.Error.ServiceUnknown reply_serial=2210
   string "The name com.wirepas.sink.sink2 was not provided by any .service files"

ERROR:root:Cannot set bytearray(b'\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff') for param cipher_key on sink sink2: GW_RES_INTERNAL_ERROR (g-dbus-error-quark: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name com.wirepas.sink.sink2 was not provided by any .service files (2))

method call time=1726052528.226912 sender=:1.28 -> destination=com.wirepas.sink.sink2 serial=2211 path=/com/wirepas/sink; interface=org.freedesktop.DBus.Properties; member=Get
   string "com.wirepas.sink.config1"
   string "AuthenticationKeySet"
error time=1726052528.243390 sender=org.freedesktop.DBus -> destination=:1.28 error_name=org.freedesktop.DBus.Error.ServiceUnknown reply_serial=2211
   string "The name com.wirepas.sink.sink2 was not provided by any .service files"

method call time=1726052528.245128 sender=:1.9672 -> destination=com.wirepas.sink.sink2 serial=26 path=/com/wirepas/sink; interface=org.freedesktop.DBus.Properties; member=Get
   string "com.wirepas.sink.config1"
   string "StackStatus"
error time=1726052528.259342 sender=org.freedesktop.DBus -> destination=:1.9672 error_name=org.freedesktop.DBus.Error.ServiceUnknown reply_serial=26
   string "The name com.wirepas.sink.sink2 was not provided by any .service files"

method call time=1726052528.260787 sender=:1.28 -> destination=com.wirepas.sink.sink2 serial=2212 path=/com/wirepas/sink; interface=com.wirepas.sink.config1; member=GetAppConfig
error time=1726052528.260975 sender=org.freedesktop.DBus -> destination=:1.28 error_name=org.freedesktop.DBus.Error.ServiceUnknown reply_serial=2212
   string "The name com.wirepas.sink.sink2 was not provided by any .service files"

ERROR:root:Cannot set Stack state. Problem in communication probably: GW_RES_INTERNAL_ERROR
Configuration done with result = GatewayResultCode.GW_RES_INTERNAL_ERROR

method call time=1726052528.314164 sender=:1.28 -> destination=com.wirepas.sink.sink2 serial=2213 path=/com/wirepas/sink; interface=org.freedesktop.DBus.Properties; member=Get
   string "com.wirepas.sink.otap1"
   string "StoredStatus"
error time=1726052528.333041 sender=org.freedesktop.DBus -> destination=:1.28 error_name=org.freedesktop.DBus.Error.ServiceUnknown reply_serial=2213
   string "The name com.wirepas.sink.sink2 was not provided by any .service files"

method call time=1726052528.394271 sender=:1.28 -> destination=com.wirepas.sink.sink2 serial=2214 path=/com/wirepas/sink; interface=org.freedesktop.DBus.Properties; member=Get
   string "com.wirepas.sink.otap1"
   string "StoredType"
error time=1726052528.399449 sender=org.freedesktop.DBus -> destination=:1.28 error_name=org.freedesktop.DBus.Error.ServiceUnknown reply_serial=2214
   string "The name com.wirepas.sink.sink2 was not provided by any .service files"

method call time=1726052528.410247 sender=:1.28 -> destination=com.wirepas.sink.sink2 serial=2215 path=/com/wirepas/sink; interface=org.freedesktop.DBus.Properties; member=Get
   string "com.wirepas.sink.otap1"
   string "StoredSeq"
error time=1726052528.414347 sender=org.freedesktop.DBus -> destination=:1.28 error_name=org.freedesktop.DBus.Error.ServiceUnknown reply_serial=2215
   string "The name com.wirepas.sink.sink2 was not provided by any .service files"

method call time=1726052528.469693 sender=:1.28 -> destination=com.wirepas.sink.sink2 serial=2216 path=/com/wirepas/sink; interface=org.freedesktop.DBus.Properties; member=Get
   string "com.wirepas.sink.otap1"
   string "StoredCrc"
error time=1726052528.489062 sender=org.freedesktop.DBus -> destination=:1.28 error_name=org.freedesktop.DBus.Error.ServiceUnknown reply_serial=2216
   string "The name com.wirepas.sink.sink2 was not provided by any .service files"

method call time=1726052528.492579 sender=:1.28 -> destination=com.wirepas.sink.sink2 serial=2217 path=/com/wirepas/sink; interface=org.freedesktop.DBus.Properties; member=Get
   string "com.wirepas.sink.otap1"
   string "StoredLen"
error time=1726052528.511888 sender=org.freedesktop.DBus -> destination=:1.28 error_name=org.freedesktop.DBus.Error.ServiceUnknown reply_serial=2217
   string "The name com.wirepas.sink.sink2 was not provided by any .service files"

method call time=1726052528.578526 sender=:1.28 -> destination=com.wirepas.sink.sink2 serial=2218 path=/com/wirepas/sink; interface=org.freedesktop.DBus.Properties; member=Get
   string "com.wirepas.sink.otap1"
   string "ProcessedSeq"
error time=1726052528.596306 sender=org.freedesktop.DBus -> destination=:1.28 error_name=org.freedesktop.DBus.Error.ServiceUnknown reply_serial=2218
   string "The name com.wirepas.sink.sink2 was not provided by any .service files"

method call time=1726052528.597919 sender=:1.28 -> destination=com.wirepas.sink.sink2 serial=2219 path=/com/wirepas/sink; interface=org.freedesktop.DBus.Properties; member=Get
   string "com.wirepas.sink.otap1"
   string "ProcessedCrc"
error time=1726052528.611061 sender=org.freedesktop.DBus -> destination=:1.28 error_name=org.freedesktop.DBus.Error.ServiceUnknown reply_serial=2219
   string "The name com.wirepas.sink.sink2 was not provided by any .service files"

method call time=1726052528.647720 sender=:1.28 -> destination=com.wirepas.sink.sink2 serial=2220 path=/com/wirepas/sink; interface=org.freedesktop.DBus.Properties; member=Get
   string "com.wirepas.sink.otap1"
   string "ProcessedLen"
error time=1726052528.673628 sender=org.freedesktop.DBus -> destination=:1.28 error_name=org.freedesktop.DBus.Error.ServiceUnknown reply_serial=2220
   string "The name com.wirepas.sink.sink2 was not provided by any .service files"

method call time=1726052528.680522 sender=:1.28 -> destination=com.wirepas.sink.sink2 serial=2221 path=/com/wirepas/sink; interface=org.freedesktop.DBus.Properties; member=Get
   string "com.wirepas.sink.otap1"
   string "FirmwareAreaId"
error time=1726052528.684981 sender=org.freedesktop.DBus -> destination=:1.28 error_name=org.freedesktop.DBus.Error.ServiceUnknown reply_serial=2221
   string "The name com.wirepas.sink.sink2 was not provided by any .service files"

method call time=1726052528.787871 sender=:1.28 -> destination=com.wirepas.sink.sink2 serial=2222 path=/com/wirepas/sink; interface=org.freedesktop.DBus.Properties; member=Get
   string "com.wirepas.sink.config1"
   string "FirmwareVersion"
error time=1726052528.788227 sender=org.freedesktop.DBus -> destination=:1.28 error_name=org.freedesktop.DBus.Error.ServiceUnknown reply_serial=2222
   string "The name com.wirepas.sink.sink2 was not provided by any .service files"

--- a/sink_service/source/config.c
+++ b/sink_service/source/config.c
@@ -312,7 +312,7 @@ static bool send_dbus_signal(const char * name)
 static int set_stack_state(sd_bus_message * m, void * userdata, sd_bus_error * error)
 {
     app_res_e res;
-    bool state;
+    int state;
     int r;
 
     /* Read the parameters */

